### PR TITLE
Add additional documentation for Batch networking and MNP

### PR DESCRIPTION
**Pull request by demartinofra**
* Added tutorial of how to run mpi with batch
* Documented network configuration with awsbatch

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/networking.rst [hunk 1]
```diff
@@ -82,5 +82,46 @@ The configuration for this architecture requires the following settings:
...
+When using ``awsbatch`` as scheduler type, ParallelCluster creates an AWS Batch Managed Compute Environment which takes
+care of managing ECS container instances, launched in the ``compute_subnet``. In order for AWS Batch to function
+correctly, Amazon ECS container instances need external network access to communicate with the Amazon ECS service
+endpoint. This translates into the following scenarios:
+
+* The ``compute_subnet`` uses a NAT Gateway to access the Internet. (Recommended approach)
+* Instances launched in the ``compute_subnet`` have public IP addresses and can reach the Internet through an
+  Internet Gateway.
+
+Additionally, according to AWS Batch docs:
```
Comment (enrico-usai) [inline docs/networking.rst]: I'd add something like: _Additionally, If you are interested on multi-node-parallel jobs ...._ (but it is my opinion, feel free to let whatever you like)

File: docs/tutorials/02_ami_customization.rst [hunk 2]
```diff
@@ -14,8 +14,8 @@ Building a custom AWS ParallelCluster AMI
     ParallelCluster. You will need to repeat the steps used to create your custom AMI with each new AWS ParallelCluster
     release.
 
-Before reading any further, take a look at the :doc:`pre_post_install` section of the documentation to determine if the
-modifications you wish to make can be scripted and supported with future AWS ParallelCluster releases
+Before reading any further, take a look at the :doc:`../pre_post_install` section of the documentation to determine
+if the modifications you wish to make can be scripted and supported with future AWS ParallelCluster releases
```
Comment (enrico-usai) [inline docs/tutorials/02_ami_customization.rst]: missing final dot.

File: docs/tutorials/03_batch_mpi.rst [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+Now that there are no jobs in the queue, we can check for output through the ``awsbout`` command. ::
+
+        $ awsbout 6efe6c7c-4943-4c1a-baf5-edbfeccab5d2
+        2018-11-12 09:41:29: Starting Job 6efe6c7c-4943-4c1a-baf5-edbfeccab5d2
+        download: s3://parallelcluster-mybatch-lui1ftboklhpns95/batch/job-hellojob_sh-1542015680924.sh to tmp/batch/job-hellojob_sh-1542015680924.sh
+        2018-11-12 09:42:00: Hello Luca from ip-172-31-4-234
+
+We can see that our job successfully ran on instance "ip-172-31-4-234".
+
+Also if you look into the /shared directory you will find a secret message for you :)
```
Comment (enrico-usai) [inline docs/tutorials/03_batch_mpi.rst]: /shared --> \`\`/shared\`\`

File: docs/tutorials/code_samples/batch_mpi/batch_hello_world.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+sleep 30
+echo "Hello $1 from $(hostname)"
+echo "Hello $1 from $(hostname)" > /shared/secret_message_for_$1_by_$AWS_BATCH_JOB_ID
```
Comment (enrico-usai) [inline docs/tutorials/code_samples/batch_mpi/batch_hello_world.sh]: I'd write `secret_message_for_${1}_by_${AWS_BATCH_JOB_ID}`

File: docs/networking.rst (hunk 1 above)
Comment (demartinofra) [reply docs/networking.rst]: I like it!

File: docs/tutorials/code_samples/batch_mpi/batch_hello_world.sh (hunk 4 above)
Comment (demartinofra) [reply docs/tutorials/code_samples/batch_mpi/batch_hello_world.sh]: 👍👍👍

File: docs/tutorials/03_batch_mpi.rst (hunk 3 above)
Comment (demartinofra) [reply docs/tutorials/03_batch_mpi.rst]: done

File: docs/tutorials/02_ami_customization.rst (hunk 2 above)
Comment (demartinofra) [reply docs/tutorials/02_ami_customization.rst]: fixed

**Pull request merged**